test/integration/scheduler_perf/scheduler_perf_test.go

```diff
@@ -0,0 +1,342 @@
+				}
+
+				nodes, err := verifyNodeCount(t, tCtx, createOp.Count)
+				if err != nil {
+					return err
+				}
+
+				strategy := createOp.NodeAllocatableStrategy
+
+				for _, node := range nodes {
+					for resourceName, expectedValue := range strategy.NodeAllocatable {
+						if node.Status.Allocatable == nil {
+							return fmt.Errorf("Node %s Status.Allocatable is nil", node.Name)
+						}
+						if allocatable, exists := node.Status.Allocatable[resourceName]; !exists {
+							return fmt.Errorf("Node %s is missing expected allocatable resource %s",
+								node.Name, resourceName)
+						} else {
+							expectedQuantity := resource.MustParse(expectedValue)
+							if allocatable.Cmp(expectedQuantity) != 0 {
+								return fmt.Errorf("Node %s has incorrect allocatable value for %s: got %s, want %s",
+									node.Name, resourceName, allocatable.String(), expectedValue)
+							}
+						}
+					}
+				}
+				return nil
+			},
+		},
+		{
+			name: "Create Nodes with Custom Template",
+			op: &createNodesOp{
+				Opcode: createNodesOpcode,
+				Count:  2,
+				NodeTemplatePath: createTempNodeTemplateFile(t, &v1.Node{
+					ObjectMeta: metav1.ObjectMeta{
+						GenerateName: "custom-node-",
+					},
+					Status: v1.NodeStatus{
+						Capacity: v1.ResourceList{
+							v1.ResourcePods:   *resource.NewQuantity(100, resource.DecimalSI),
+							v1.ResourceCPU:    resource.MustParse("4"),
+							v1.ResourceMemory: resource.MustParse("8Gi"),
+						},
+						Phase: v1.NodeRunning,
+						Conditions: []v1.NodeCondition{
+							{Type: v1.NodeReady, Status: v1.ConditionTrue},
+						},
+					},
+				}),
+			},
+			verifyFunc: func(t *testing.T, tCtx ktesting.TContext, op realOp) error {
+				createOp, ok := op.(*createNodesOp)
+				if !ok {
+					return fmt.Errorf("Expected createNodesOp but got %T", op)
+				}
+
+				nodes, err := verifyNodeCount(t, tCtx, createOp.Count)
+				if err != nil {
+					return err
+				}
+
+				for _, node := range nodes {
+					podCapacity := node.Status.Capacity[v1.ResourcePods]
+					expectedPodCapacity := resource.MustParse("100")
+					if podCapacity.Cmp(expectedPodCapacity) != 0 {
+						return fmt.Errorf("Node %s has incorrect pod capacity: got %s, want %s",
+							node.Name, podCapacity.String(), expectedPodCapacity.String())
+					}
+
+					cpuCapacity := node.Status.Capacity[v1.ResourceCPU]
+					expectedCPUCapacity := resource.MustParse("4")
+					if cpuCapacity.Cmp(expectedCPUCapacity) != 0 {
+						return fmt.Errorf("Node %s has incorrect CPU capacity: got %s, want %s",
+							node.Name, cpuCapacity.String(), expectedCPUCapacity.String())
+					}
+
+					isReady := false
+					for _, condition := range node.Status.Conditions {
+						if condition.Type == v1.NodeReady && condition.Status == v1.ConditionTrue {
+							isReady = true
+							break
+						}
+					}
+					if !isReady {
+						return fmt.Errorf("Node %s is not in Ready condition as expected", node.Name)
+					}
```

Comment:
It could be a separate func so that we can use it for other op's tests later?

```go
// verifyObj checks if listed objects are the same as passed expected object.
func verifyObj(expected any) func (t *testing.T, tCtx ktesting.TContext, op realOp) error 
```